dwds/debug_extension/web/background.dart

```
@@ -93,14 +111,56 @@ void main() {
       callback(List.from(tabs));
     }));
   });
-  addListener(startDebug);
+  addListener(startDebugging);
 
   // For testing only.
   onFakeClick = allowInterop(() {
-    startDebug(null);
+    startDebugging(null);
   });
 
   isDartDebugExtension = true;
+
+  onMessageExternalAddListener(allowInterop(
+      (Request request, Sender sender, Function sendResponse) async {
+    if (_allowedExtensions.contains(sender.id)) {
+      if (request.name == 'sendCommand') {
+        try {
+          var options = request.options as SendCommandOptions;
+          sendCommand(Debuggee(tabId: request.tabId), options.method,
+              options.commandParams, allowInterop(([e]) {
+            // No arguments indicate that an error occurred.
+            if (e == null) {
+              sendResponse(ErrorResponse()..error = stringify(lastError));
+            } else {
+              sendResponse(e);
+            }
+          }));
+        } catch (e) {
+          sendResponse(ErrorResponse()..error = '$e');
+        }
+      } else if (request.name == 'encodedUri') {
+        sendResponse(_tabIdToEncodedUri[request.tabId]);
+      } else if (request.name == 'startDebugging') {
+        startDebugging(null);
+      } else {
+        sendResponse(
+            ErrorResponse()..error = 'Unknown request name: ${request.name}');
+      }
+    }
+  }));
+}
+
+void sendMessageToExtensions(Request request) {
+  for (var extensionId in _allowedExtensions) {
+    try {
+      sendMessage(extensionId, request, RequestOptions(), allowInterop(([e]) {
+        if (e == null) {
+          // Error sending message. Check lastError to silently fail.
+          lastError;
+        }
+      }));
+    } catch (_) {}
```

avoid silently swallowing exceptions if possible

---

I suppose we can log to the console since that's usually hidden from the user anyway. May be helpful for debugging purposes.